Foundation/Host.swift

```diff
@@ -138,7 +140,12 @@ open class Host: NSObject {
                 }
                 let sa_len: socklen_t = socklen_t((family == AF_INET6) ? MemoryLayout<sockaddr_in6>.size : MemoryLayout<sockaddr_in>.size)
                 let lookupInfo = { (content: inout [String], flags: Int32) in
-                    if getnameinfo(info.ai_addr, sa_len, host, socklen_t(NI_MAXHOST), nil, 0, flags) == 0 {
+                    #if os(Android)
+                    typealias maxhostlen_t = Int
+                    #else
+                    typealias maxhostlen_t = socklen_t
+                    #endif
+                    if getnameinfo(info.ai_addr, sa_len, host, maxhostlen_t(NI_MAXHOST), nil, 0, flags) == 0 {
```

Comment:
What is going on here? The fourth argument should be of type `size_t` on all platforms, should it not? I'm not aware of any type known as `maxhostlen_t`. If necessary, you can simply use `numericCast(NI_MAXHOST)` on all platforms.